R/MixedModelsCommon.R

```diff
@@ -792,1293 +793,4580 @@
+    )
+  }
+  if (options$plotsBackgroundColor != "none" && options$plotsGeom != "geom_jitter" && "color" %in% mapping)
+    data_arg$color <- options$plotsBackgroundColor
+
+  # fixing afex issues with bootstrap and LRT type II SS - hopefully removeable in the future
+  if (type %in% c("LMM", "GLMM"))
+    if (options$method %in% c("LRT", "PB") && options$type == 2)
+      model <- model$full_model[[length(model$full_model)]]
+
+  .setSeedJASP(options)
+  p <- tryCatch(
+    afex::afex_plot(
+      model,
+      dv          = .v(options$dependentVariable),
+      x           = .v(unlist(options$plotsX)),
+      trace       = if (length(options$plotsTrace) != 0) .v(unlist(options$plotsTrace)),
+      panel       = if (length(options$plotsPanel) != 0) .v(unlist(options$plotsPanel)),
+      id          = .v(options$plotsAgregatedOver),
+      data_geom   = getFromNamespace(options$plotsGeom, geom_package),
+      mapping     = mapping,
+      error       = options$plotsCImethod,
+      error_level = options$plotsCIwidth,
+      data_alpha  = options$plotAlpha,
+      data_arg    = if (length(data_arg) != 0) data_arg,
+      error_arg   = list(
+        width = 0,
+        size  = .5 * options$plotRelativeSize
+      ),
+      point_arg   = list(size = 1.5 * options$plotRelativeSize),
+      line_arg    = list(size = .5 * options$plotRelativeSize),
+      legend_title = paste(.unv(unlist(options$plotsTrace)), collapse = "\n"),
+      dodge       = options$plotDodge
+    ),
+    error = function(e)
+      e
+  )
+
+  if (any(class(p) %in% c("simpleError", "error"))) {
+    plots$setError(p$message)
+    return()
+  }
+
+  if (options$plotsGeom == "geom_violin" && (length(options$plotsAgregatedOver) == 1 && length(unique(dataset[, .v(options$plotsAgregatedOver)])) < 3)) {
+    plots$setError(gettext("Violin geom requires that the random effects grouping factors has at least 3 levels."))
+    return()
+  }
+
+  # fix the axis
+  p <- .mmFixPlotAxis(p)
+
+  # fix names of the variables
+  p <- p + ggplot2::labs(x = unlist(options$plotsX), y = options$dependentVariable)
+
+  # add theme
+  if (options$plotsTheme == "JASP") {
+
+    p <- jaspGraphs::themeJasp(p, legend.position = options$plotLegendPosition)
+
+  } else  if (options$plotsTheme != "JASP") {
+
+    p <- p + switch(
+      options$plotsTheme,
+      "theme_bw"      = ggplot2::theme_bw()       + ggplot2::theme(legend.position = "bottom"),
+      "theme_light"   = ggplot2::theme_light()    + ggplot2::theme(legend.position = "bottom"),
+      "theme_minimal" = ggplot2::theme_minimal()  + ggplot2::theme(legend.position = "bottom"),
+      "theme_pubr"    = jaspGraphs::themePubrRaw(legend = options$plotLegendPosition),
+      "theme_apa"     = jaspGraphs::themeApaRaw(legend.pos = switch(
+        options$plotLegendPosition,
+        "none"  = "none",
+        "botom" = "bottommiddle",
+        "right" = "bottomright",
+        "top"   = "topmiddle",
+        "left"  = "bottomleft"
+      ))
```

Comment:
1. Are you sure that `botom` shouldn't be `bottom`?
2. I find that it leards to nicer code if the values in QML for `$plotLegendPosition` correspond directly to the values that R expects, instead of adding an additional translation step in between. Basically, can't the values for the QML element correspond to `"none"`, `"bottom"`, etc. and then you can delete this switch?

Reply:
1) thanks, it was a typo, fixed

2) it does for all other themes, the APA one does not have those positions